src/packet/key/secret.rs

```diff
@@ -38,48 +38,7 @@ impl<D> Drop for SecretKeyInner<D> {
     }
 }
 
-/// Altering the password protection state of a Secret Key packet.
-/// Removing or adding password protection for the private key material in a Secret Key packet.
-///
-/// See https://www.rfc-editor.org/rfc/rfc4880.html#section-3.7.2.1
-pub trait SecretKeyProtection {
```

Comment:
Final bit of musing on the trait: I had wondered if/hoped that this trait might enable users of the library to abstract over `SecretKey` and `SecretSubkey`. But am unsure if that would actually ever be useful in practice.